storage_service/static/js/project.js

```diff
@@ -1,7 +1,54 @@
-$(document).ready(function () {
+var getAjaxDataTableOptions = function (options, ajaxSource, headers, filter) {
+  var result = {};
+  for (var k in options) {
+    result[k] = options[k];
+  }
+  // enable server-side processing
+  result["bServerSide"] = true;
+  result["bProcessing"] = true;
+  result["sAjaxSource"] = ajaxSource;
+  if (undefined !== filter) {
```

Comment:
I think this is the first Yoda condition I've seen in JS! I did a double-take but I think it's fine to keep as-is.